### PR TITLE
fix_db_inconsistencies

### PR DESCRIPTION
**Pull request by jmcgettrick**

### CODE REVIEW

File: db/upgrade.php [hunk 1]
```diff
@@ -253,5 +253,41 @@ function xmldb_turnitintooltwo_upgrade($oldversion) {
...
+
+        $table = new xmldb_table('turnitintooltwo_parts');
+        // Ensure unanon length is 1.
+        $field = new xmldb_field('unanon', XMLDB_TYPE_INTEGER, '1', null, XMLDB_NOTNULL, null, 0, 'migrated');
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->change_field_precision($table, $field);
+        }
+        // Ensure submitted length is 1.
+        $field = new xmldb_field('submitted', XMLDB_TYPE_INTEGER, '1', null, XMLDB_NOTNULL, null, 0, 'unanon');
+        if (!$dbman->field_exists($table, $field)) {
```
Comment (golenkovm) [inline db/upgrade.php:280]: Looks like a type.
Should this be like `if ($dbman->field_exists($table, $field)) {`?

File: db/upgrade.php [hunk 2]
```diff
@@ -253,5 +253,41 @@ function xmldb_turnitintooltwo_upgrade($oldversion) {
...
+            $dbman->change_field_precision($table, $field);
+        }
+
+        $table = new xmldb_table('turnitintooltwo_users');
+        // Ensure user_agreement_accepted allows null.
+        $field = new xmldb_field('user_agreement_accepted', XMLDB_TYPE_INTEGER, '1', false, false, null, 0, 'instructor_rubrics');
+        if (!$dbman->field_exists($table, $field)) {
```
Comment (golenkovm) [inline db/upgrade.php:287]: Same here.
`if ($dbman->field_exists($table, $field)) {`

**Comment by golenkovm** [comment]
Hi @jmcgettrick 
Thanks for working on the issue. All looks good to me.
Just two tiny bits with `!`.
Cheers

File: db/upgrade.php (hunk 1 above)
Comment (jmcgettrick) [reply db/upgrade.php]: Thanks @golenkovm

File: db/upgrade.php (hunk 2 above)
Comment (jmcgettrick) [reply db/upgrade.php]: Thanks @golenkovm

**Pull request merged**